src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -143,18 +155,29 @@ def load_command_table(self, command_loader):
                 _parser=command_parser)
 
     def validation_error(self, message):
-        telemetry.set_user_fault('validation error: {}'.format(message))
-        return super(AzCliCommandParser, self).error(message)
+        az_error = AzCLIError(AzCLIErrorType.ValidationError, message, command=self.prog)
+        az_error.print_error()
+        az_error.send_telemetry()
+        self.exit(2)
 
     def error(self, message):
-        telemetry.set_user_fault('parse error: {}'.format(message))
-        args = {'prog': self.prog, 'message': message}
-        with CommandLoggerContext(logger):
-            logger.error('%(prog)s: error: %(message)s', args)
-        self.print_usage(sys.stderr)
-        # Manual recommendations
-        self._set_manual_recommendations(args['message'])
-        # AI recommendations
+        # Get a recommended command from the CommandRecommender
+        command_arguments = self._get_failure_recovery_arguments()
+        recommender = CommandRecommender(*command_arguments, self.cli_ctx)
+        recommender.set_help_examples(self.get_examples(self.prog))
+        recommendation = recommender.recommend_a_command()
```

Comment:
Won't this be a duplication of `ai-did-you-mean-this`?

Reply:
Has been discussing with Aladdin team, will make it retired if necessary.